### PR TITLE
The installed_pkgs.log file now covers dynamic builddeps

### PR DESCRIPTION
**Pull request by praiskup**
This change required us to create a new "postdeps" hook with post-%generate_buildrequires timing.

Fixes: #1429
Relates: #1393

### CODE REVIEW

File: mock/py/mockbuild/plugins/package_state.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def __init__(self, plugins, conf, buildroot):
         self.inst_done = False
         self.online = self.buildroot.config['online']
         plugins.add_hook("postyum", self._availablePostYumHook)
-        plugins.add_hook("prebuild", self._installedPreBuildHook)
```
Comment (xsuchy) [inline mock/py/mockbuild/plugins/package_state.py:46]: Although this hook is not documented, it is used by several plugins. I believe it should not be removed.

Comment (praiskup) [reply mock/py/mockbuild/plugins/package_state.py:46]: The hooks in this "prebuild" namespace are still called, this particular hook is just moved to a later timing...

Comment (xsuchy) [reply mock/py/mockbuild/plugins/package_state.py:46]: Right. I was confused and thought you are replacing the hook itself. Sorry for the noise.

**Comment by xsuchy** [comment]
+1

**Pull request merged**